addons/blocklydevtools/userscript.js

```diff
@@ -0,0 +1,419 @@
+                throw new Error("Workspace XML does not contain block!");
+            }
+            xmlStr = xmlStr.replace(originalXml, container.textContent); //Replace original XML with modified XML.
+            //xmlStr = xmlStr.replace(/\u2003/g, "").replace(/\n/g, "");
+            xmlStr = xmlStr.replace(/\u00A0/g, "\u0020"); //Replace any non-breaking spaces with normal ones.
+            dom = (Blockly.Xml.textToDom || Blockly.utils.xml.textToDom)(xmlStr); //Update the DOM variable.
+            workspace.clear(); //Clear the workspace
+            Blockly.Xml.domToWorkspace(dom, workspace); //Load the DOM
+            if (Blockly.getMainWorkspace().getToolbox() //If the blockly instance has a toolbox, it needs to be refreshed,
+            ) {
+                Blockly.getMainWorkspace().getToolbox().refreshSelection();
+            }
+        }
+
+        //Prevent events from propagating to Blockly
+        container.addEventListener("pointerdown", (event) => {
+            event.stopPropagation();
+        }, {
+            capture: true
+        });
+        container.addEventListener("wheel", (event) => {
+            event.stopPropagation();
+        }, {
+            capture: true
+        });
+        container.addEventListener("scroll", (event) => {
+            event.stopPropagation();
+        }, {
+            capture: true
+        });
+        container.addEventListener("contextmenu", (event) => {
+            event.stopPropagation();
+        }, {
+            capture: true
+        });
+        container.addEventListener("keydown", (event) => {
+            TabManager.enableTab(event, "\u2003"); //Fix for tab key
+        }, {
+            capture: true
+        });
+
+        return container;
+    }
+    function processBlock(element) { //Function to hook block svg element
+        if (isNewBlocklyBlock(element)) { //Check if we have not already hooked it
+            var blockId = element.getAttribute("data-id"); //Get the id
+            var internalBlock = workspace.getBlockById(blockId); //Get the internal block object
+            internalBlock.tooltip ||= internalBlock.type || "unknown"; //If the block does not have a tooltip, set it to it's opcode.
+            var devWrapper = document.createElementNS('http://www.w3.org/2000/svg', 'foreignObject'); //Create a foreignObject element to allow HTML inside of SVG.
+            var btnWrapper = document.createElement("div"); //Wrapper for buttons to keep them on the same row
+
+            btnWrapper.style.cursor = "auto"; //Fix for cursor being the grab hand
+            btnWrapper.style.width = "max-content";
+
+            var btn = document.createElement("span"); //Create the edit button
+            btn.style.cursor = "pointer";
+            btn.style.zIndex = "999";
+            btn.style.lineHeight = "1rem";
+            btn.setAttribute("data-is-blocklydev-btn", "true");
+            btn.innerHTML = "✏️"; //📝✏️
```

Comment:
Ideally all these emojis would be proper icons instead I don't think that has to be done now.